x-pack/plugins/cloud_security_posture/public/pages/findings/findings_container.tsx

```diff
@@ -5,66 +5,107 @@
  * 2.0.
  */
 
-import React, { useCallback, useEffect, useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import { css } from '@emotion/react';
 import { EuiSpacer } from '@elastic/eui';
-import styled from 'styled-components';
-import { DataView } from '../../../../../../src/plugins/data/common';
+import type { SearchResponse } from '@elastic/elasticsearch/lib/api/types';
+import type { UseMutationResult } from 'react-query';
+import type { Filter } from '@kbn/es-query';
 import { FindingsTable } from './findings_table';
-import { useKibana } from '../../../../../../src/plugins/kibana_react/public';
+import { FindingsRuleFlyout } from './findings_flyout';
 import { FindingsSearchBar } from './findings_search_bar';
-import { CSP_KUBEBEAT_INDEX_PATTERN } from '../../../common/constants';
-import type { CSPFinding, FetchState } from './types';
-import type { CspPluginSetup } from '../../types';
+import { TEST_SUBJECTS } from './constants';
+import {
+  extractErrorMessage,
+  useSourceQueryParam,
+  useEsClientMutation,
+  isNonNullable,
+} from './utils';
+import type { CSPFinding, FindingsFetchState } from './types';
+import type { DataView, IKibanaSearchResponse } from '../../../../../../src/plugins/data/common';
+import type { SearchBarProps } from '../../../../../../src/plugins/data/public';
+import { useKibana } from '../../../../../../src/plugins/kibana_react/public';
+
+type FindingsEsSearchMutation = UseMutationResult<
+  IKibanaSearchResponse<SearchResponse<CSPFinding>>,
+  unknown,
+  void
+>;
+
+export type URLState = Parameters<NonNullable<SearchBarProps['onQuerySubmit']>>[0] & {
+  filters: Filter[];
+};
+
+const getDefaultQuery = (): Required<URLState> => ({
+  query: { language: 'kuery', query: '' },
+  filters: [],
+  dateRange: {
+    from: 'now-15m',
+    to: 'now',
+  },
+});
+
+// with https://github.com/microsoft/TypeScript/pull/46266
+// destructuring would make this more concise and won't leak props
+export const getFetchState = <T extends FindingsEsSearchMutation>(v: T): FindingsFetchState => {
+  switch (v.status) {
+    case 'error':
+      return { ...v, error: extractErrorMessage(v.error) };
+    case 'success':
+      return {
+        ...v,
+        data: v.data?.rawResponse?.hits?.hits?.map((h) => h._source).filter(isNonNullable),
+      };
+    default:
+      return v;
+  }
+};
 
 /**
  * This component syncs the FindingsTable with FindingsSearchBar
  */
-export const FindingsTableContainer = () => {
-  const { kubebeatDataView } = useKubebeatDataView();
-  const [state, set] = useState<FetchState<CSPFinding[]>>({
-    loading: false,
-    error: false,
-    data: [],
+export const FindingsTableContainer = ({ dataView }: { dataView: DataView }) => {
+  const { notifications } = useKibana().services;
+  const [selectedFindingsItem, selectItem] = useState<CSPFinding | undefined>();
```

Comment:
```suggestion
  const [selectedFinding, setSelectedFinding] = useState<CSPFinding | undefined>();
```

a few things:
1. I think the `set` is important it makes it easier to understand that this function is going to cause renders when reading it further down, also of course it's conventional.
2. the type is correctly called `finding` and not `findingsitem`, I think naming it the same is helping keep everything on the same line.

Reply:
good call! makes it more consistent 

added manually as it requires more changes for consumers